README.md
# pkf_trackpoint
Configure TrackPoint on GNU/Linux.  This menu-driven script allows the user to
  - Adjust sensitivity, speed, and touch-to-click on the fly (not persistent between reboots)
  - If distribution uses 'evdev' for xinput (such as RHEL/CentOS), toggle TrackPoint scrolling
  - Configure boot script to apply settings on boot (persistent between reboots)
  - Remove boot scripts to restore OS to original configuration

## Usage
### Run "pkf_trackpoint.sh" to configure trackpoint.
  1. Displays current settings and persistence state
  2. Apply trackpoint settings on the fly
  3. Toggle TrackPoint scrolling (only for GNU/Linux distos using "evdev")
  4. Enable persistent settings (using current settings).  User may then manually edit /usr/bin/trackpoint.sh as they desire
  5. Make current settings persistent
  6. Remove all pkf_trackpoint files from initialization daemon
  7. Sets trackpoint back to OS defaults (sensitivity=128, speed=97, press=0)
  0. Exit pkf_trackpoint

### Run "testing.sh" to check status and enable/disable test mode (for development)
Turning on test mode will redirect configuration changes to the embedded "testing" directory.

This can be useful for understanding what pkf_trackpoint is doing as well as assist in testing any changes
  - Displays current state of test mode.
  - Displays whether fake trackpad exists
  1. Switch between testing mode being on or off
  2. Toggles whether a fake trackpad exists or not
  0. Exit

## Files created/modified
  - Files restored every boot (do not delete)
    + /sys/devices/<...>/sensitivity
    + /sys/devices/<...>/speed
    + /sys/devices/<...>/press_to_select
  - Files that remain between reboots (if using persistent settings)
    + /etc/systemd/system/trackpoint.service (systemd)
    + /etc/systemd/system/trackpoint.timer (systemd)
    + /etc/init.d/trackpoint (SysV)
    + /usr/bin/trackpoint.sh
    + /<usr|etc>/.../xorg.conf.d/90-trackpoint.conf

## Should work on...
  - Ubuntu/Kubuntu (15.04 and later)
  - OpenSUSE Leap & Tumbleweed
  - Trisqel
  - KDE Neon
  - CentOS

## Exit codes
  - 100: Script not run as root (required since system file are being changed)
  - 200: Trackpoint not detected (or at least common Trackpoint sys directories do not appear to exist.)

## Variables used
  - file = temporary variable used in FOR loop to capture locations of files named "press_to_select" located in /sys/devices directory
  - dir = temporary variable used in FOR loop to capture the parent directory of each $file found
  - root_dir = temporary variable used in FOR loop to capture starting directory for finding xorg.conf.d (value used for vXorgDir)
  - vTrackpointPath = Path to Trackpoint settings. Varies based on existence of a trackpad ( sys/devices/platform/i8042/serio1 | sys/devices/platform/i8042/serio1/serio2 )
  - vInitSystem = Detected initialization service used ( sysv | systemd )
  - vInitStatus = Current status of persistence ( Enabled | Disabled | Broken. Use Option 3 or 5)
  - vMainMenu = Selected option from menu ( [0-6] )
  - vTempSensitivity = Pre-vallidation value for sensitivity setting ( [1-255] )
  - vTempSpeed = Pre-vallidation value for speed setting ( [1-255] )
  - vTempPress_to_Select = Pre-vallidation value for press_to_select setting ( [0-1] )
  - vSensitivity = Validated value for sensitivity setting ( [1-255] )
  - vSpeed = Validated value for speed setting ( [1-255] )
  - vPress_to_Select = Validated value for press_to_select setting ( [0-1] )
  - vScrolling = Track user selection for TrackPoint scrolling, defaults to "9". "0" is disabled, "1" is enabled. ( [0-1] | 9)
  - vTrackpointXconfig = Holds data if using evdev for xinput, empty if not. (<some output> | <empty> )  
  - vXorgDir = Locatoin of xorg.conf.d for setting scrolling persistence (<xorg.conf.d path> | <empty>)
  - vTrackpointWheelEnabledValue = Current value of "Evdev EmulateWheel. "0" is disabled, "1" is enabled. ([0-1] | <empty>)
  - vTrackpointWheelButtonValue  Current value of "Evdev EmulateWheelButton".  Anything except "2" is in a "disabled" state ([2-4])

## TODO
  - Add support for SysV (non-Debian) & Upstart
  - Find better initialization than timer for systemd


CHANGE_LOG.md
- v0.0.1:
  + Non-working initial save
- v0.0.2
  + Non-working checkpoint
  + Configure "testing" & "template" directories
  + Add variable null check to option 4
  + Change exit case to "0"
  + Add actual configuration to systemd files
##### v1.0.0
  + Remove redirections to testing directory
  + Change main script's name
- v1.0.1
  + Uncomment systemctl commands
- v1.1.0
  + Add toggle to enable/disable test mode
- v1.1.1
  + Fix README file
- v1.1.2
  + Fix README file
  + Disable test mode
##### v1.2.0
  + Additional info in "Check Settings"
  + Streamline newlines for main menu for pkf_trackpoint and testing
  + Add "Remove persistence" option
  + Update README
- v1.2.1
  + Minor README fix
##### v1.3.0
  + Add compatability with no/disabled trackpad devices
  + Add testing toogle to add/remove trackpad paths.
- v1.3.1
  + Remove trackpad-less TODO from README
  + Add systemd reload when enabling persistence
- v1.3.2
  + Fix typo in setting persistence
  + Set trackpoint.sh initial template to OS defaults
  + Update README with tested laptops
- v1.3.3
  + Fix testing.sh for trackpad testing
  + Change testing.sh trackpoint settings to OS defaults
- v1.3.4
  + Add systemctl stop/daemon-reload to persistence removal
- v1.3.5
  + Fix typo in systemctl stop for persistence removal
##### v1.4.0
  + Add reset current Trackpoint settings to defaults
##### v1.5.0
  + Use path unit instead of timer unit
  + Add conditional option to remove pre-v1.5 settings
- v1.5.1
  + Fix type in timer removal in option 5
  + Change timer to path in persistence check and option 3
  + Change "enable" to "trackpoint" in option 3
- v1.4.1
  + Revert back to v1.4
  + Add condition triggers for systemd service
- v1.4.2
  + Disable testing mode (left enabled by mistake)
##### v1.5.0
  + Removed due to missing features for systemd's path unit
##### v1.6.0
  + Add Sys V init for Debian based distros
  + Add complete addition/removal of testing directory in test toggle
- v1.6.1
  + Make Sys V init only run at level 5
  + Add LSB info to Sys V script
  + Update menu text in pkf_trackpoint
- v1.6.2
  + Allow output of persistence setting but hide systemd service stop (non-applicable warning about being called by timer)
##### v1.7.0
  + Enable stop/restart for persistence settings
  + Add conditional to trackpad detection for if neither exists (exit 200)
  + Reset default setting values in trackpoint.sh when disabling test mode
- v1.7.1
  + Add missing command to apply changes if re-running Option 4 (applying current settings to persistence )
  + Fix SysV output being hidden when using Option 3 (setting up persistence)
  + Update CHANGELOG
- v1.7.2
  + Remove command to start trackpoint.sh in Option 4 (redundant since settings already applied in Option 2)
- v1.7.3
  + Add initially tested OS/Laptops to README
##### v1.8.0
  + Add check for root user
  + Add validation of setting values
  + Prevent using values entered if any are not valid
  + Display current corresponding trackpoint setting when setting new values
  + Add better detection of trackpoint existence
  + Initialize trackpoint variables with current setting values
  - v1.8.1
    + Fix missing validation of persistence before applying option 4 (apply current settings to persistence)
    + Fix redirection of systemd warnings when disabling persistence (can be safely ignored as timer unit is also deleted)
    + Tweak persistence detection and reporting
    + Change systemd timer from "OnBootSec" to "OnActiveSec"
  - v1.8.2
    + Add trackpoint path for KDE Neon
  - v1.8.3
    + Fix typo in KDE Neon support
    + Add systemd conditional for KDE Neon
  - v1.8.4
    + Update template with current settings at start and on setting (Option 2) to prevent settings being undone when setting persistence (Option 3)
  - v1.8.5
    + Throw error if user attempts to disable persistence if it's not enabled
    + Apply current settings upon disabling persistence (to prevent setting changes)    
##### v1.9.0
  + Change Trackpoint configuration directory assignment to dynamic discovery instead of hard-coded conditional paths
  + Remove conditional path existence check in trackpoint.service since it's redundant given the check in trackpoint.sh
#### v2.0.0
  + Add functionality to enable/disable scrolling for evdev (libinput generally already has good scrolling).
  + Version bump due to scenario where new file (90-trackpoint.conf) written to system with this version and cannot be removed via older versions.
  + Minor polish of code (mostly condensing "if;then" to single line)
  - v2.0.1
    + Fix erroneous leftovers from non-evdev scrolling work
    + Fix missing space in conditional for scrolling option


run.sh
#!/bin/bash

while :
do

echo "1: Check current Trackpoint settings"
echo "2: Set Trackpoint settings"
echo "3: Setup systemd for persistent Trackpoint settings"
echo "4: Make current Trackpoint settings persistent"
echo "0: Exit"
echo " "
read vMainMenu

case $vMainMenu in
1 )
  printf "Trackpoint Sensitivity: " && cat testing/sys/devices/platform/i8042/serio1/serio2/sensitivity
  printf "Trackpoint Speed: " && cat testing/sys/devices/platform/i8042/serio1/serio2/speed
  printf "Trackpoint Press_To_Select: " && cat testing/sys/devices/platform/i8042/serio1/serio2/press_to_select
  echo " "
;;

2 )
  echo "Sensitivity:1-255"
  read vSensitivity
  echo "Speed:1-255"
  read vSpeed
  echo "Press_to_Select:0-1"
  read vPress_to_Select
  echo $vSensitivity > testing/sys/devices/platform/i8042/serio1/serio2/sensitivity
  echo $vSpeed > testing/sys/devices/platform/i8042/serio1/serio2/speed
  echo $vPress_to_Select > testing/sys/devices/platform/i8042/serio1/serio2/press_to_select
  echo " "
;;

3 )
  cp -r templates/trackpoint.service testing/etc/systemd/system
  cp -r templates/trackpoint.timer testing/etc/systemd/system
  cp -r templates/trackpoint.sh testing/usr/bin && chmod +x testing/usr/bin/trackpoint.sh
  ##Uncomment systemctl commands after test validation
  #systemctl start trackpoint
  #systemctl enable trackpoint.timer
;;

4 )
if [ ! -z "$vSensitivity" -a ! -z "$vSpeed" -a ! -z "$vPress_to_Select" ];
then
  echo "#!/bin/bash" > templates/trackpoint.sh
  echo "echo -n $vSensitivity > /sys/devices/platform/i8042/serio1/serio2/sensitivity" >> templates/trackpoint.sh
  echo "echo -n $vSpeed > /sys/devices/platform/i8042/serio1/serio2/speed" >> templates/trackpoint.sh
  echo "echo -n $vPress_to_Select > /sys/devices/platform/i8042/serio1/serio2/press_to_select" >> templates/trackpoint.sh
  cp -r templates/trackpoint.sh testing/usr/bin && chmod +x testing/usr/bin/trackpoint.sh
else
  echo "Settings not configured.  Please run \"Option 2:\" first"
  echo ""
fi
;;

0 | [eE]xit )
  exit
;;

* )
  echo "Pleaes enter a valid option"
  echo " "
esac

done


testing.sh
#!/bin/bash

while :
do

echo ""
grep "testing/" pkf_trackpoint.sh > /dev/null || grep "testing/" templates/trackpoint.sh > /dev/null
if [ $? -gt 0 ];
then
  echo "Test mode: False"
  vTestStatus=0
else
  echo "Test mode: True"
  vTestStatus=1
fi

if [ ! -d testing/sys/devices/platform/i8042/serio1/serio2 ];
then
  echo "Trackpad: Disabled"
  vTestTrackpad=0
else
  echo "Trackpad: Enabled"
  vTestTrackpad=1
fi

echo ""
echo "1: Toogle Test Status"
echo "2: Toogle Trackpad existence"
echo "0: Exit"
echo " "
read vTestMenu

case $vTestMenu in
  1 )
  if [ $vTestStatus = 0 ]; #If test mode is disabled, enable test mode
  then
    sed -i 's/\/sys\/devices/testing\/sys\/devices/g' pkf_trackpoint.sh  #Set "/sys/devices" path to test mode
    sed -i 's/\/etc\/systemd/testing\/etc\/systemd/g' pkf_trackpoint.sh  #Set "/etc/systemd" path to test mode
    sed -i 's/\/etc\/init.d/testing\/etc\/init.d/g' pkf_trackpoint.sh  #Set "/etc/init.d" path to test mode
    sed -i 's/\/usr\/bin/testing\/usr\/bin/g' pkf_trackpoint.sh  #Set "/usr/bin" path to test mode
    sed -i 's/systemctl/#systemctl/g' pkf_trackpoint.sh  #Comment out systemd commands
    sed -i 's/#systemctl --version/systemctl --version/g' pkf_trackpoint.sh  #Reset the command for systemd check
    sed -i 's/update-rc.d/#update-rc.d/g' pkf_trackpoint.sh  #Comment out update-rc commands
    sed -i 's/\/sys\/devices/testing\/sys\/devices/g' templates/trackpoint.sh  #Set "/sys/devices" path to test mode
    mkdir -p testing/etc/systemd/system
    mkdir -p testing/etc/init.d
    mkdir -p testing/usr/bin
    mkdir -p testing/sys/devices/platform/i8042/serio1
    echo "128" > testing/sys/devices/platform/i8042/serio1/sensitivity
    echo "92" > testing/sys/devices/platform/i8042/serio1/speed
    echo "0" > testing/sys/devices/platform/i8042/serio1/press_to_select
  fi

  if [ $vTestStatus = 1 ]; #If test mode is enabled, disable test mode
  then
    sed -i 's/testing\/sys\/devices/\/sys\/devices/g' pkf_trackpoint.sh  #Unset "/sys/devices" path from test mode
    sed -i 's/testing\/etc\/systemd/\/etc\/systemd/g' pkf_trackpoint.sh  #Unset "/etc/systemd" path from test mode
    sed -i 's/testing\/etc\/init.d/\/etc\/init.d/g' pkf_trackpoint.sh  #Unset "/etc/init.d" path from test mode
    sed -i 's/testing\/usr\/bin/\/usr\/bin/g' pkf_trackpoint.sh  #Unset "/usr/bin" path from test mode
    sed -i 's/#systemctl/systemctl/g' pkf_trackpoint.sh  #Uncomment systemd commands
    sed -i 's/#update-rc.d/update-rc.d/g' pkf_trackpoint.sh  #Uncomment update-rc commands
    sed -i 's/testing\/sys\/devices/\/sys\/devices/g' templates/trackpoint.sh  #Unset "/sys/devices" path from test mode
    sed -i "/vSensitivity=/c\vSensitivity=128" templates/trackpoint.sh
    sed -i "/vSpeed=/c\vSpeed=92" templates/trackpoint.sh
    sed -i "/vPress_to_Select=/c\vPress_to_Select=0" templates/trackpoint.sh
    rm -rf testing
  fi
  exit
  ;;

  2 )
  if [ $vTestTrackpad = 0 ];
  then
    mkdir -p testing/sys/devices/platform/i8042/serio1/serio2
    echo "128" > testing/sys/devices/platform/i8042/serio1/serio2/sensitivity
    echo "92" > testing/sys/devices/platform/i8042/serio1/serio2/speed
    echo "0" > testing/sys/devices/platform/i8042/serio1/serio2/press_to_select
  fi

  if [ $vTestTrackpad = 1 ];
  then
    rm -rf testing/sys/devices/platform/i8042/serio1/serio2
  fi
  exit
  ;;

0 | [eE]xit )
  exit
  ;;

* )
  echo "Pleaes enter a valid option"
  echo " "
esac

done


templates/trackpoint.sh
#!/bin/bash

## Determine and set correct path for trackpoint
for file in $(find /sys/devices -name press_to_select)
do
  dir=$(dirname $file)
  if [ -f $dir/speed -a -f $dir/sensitivity ]; then
    vTrackpointPath=$dir
    break
  fi
done

if ! [ $vTrackpointPath ]; then
  echo "Trackpoint not detected"
  exit 200
fi

## Set trackpoint setting variable values
vSensitivity=128
vSpeed=97
vPress_to_Select=0

## Apply settings based on option "start" or "stop" being passed
case $1 in
  [Ss]tart )
    echo -n $vSensitivity > $vTrackpointPath/sensitivity
    echo -n $vSpeed > $vTrackpointPath/speed
    echo -n $vPress_to_Select > $vTrackpointPath/press_to_select
  ;;

  [Ss]top )
    echo 128 > $vTrackpointPath/sensitivity
    echo 97 > $vTrackpointPath/speed
    echo 0 > $vTrackpointPath/press_to_select
  ;;

  * )
    echo "Usage: {start|stop}"
esac


pkf_trackpoint.sh
#!/bin/bash

## Ensure script is being run as root
if ! [ $(id -u) = 0 ]; then
   echo "Pleaes run this script as root"
   exit 100
fi

## Determine and set correct path for trackpoint
for file in $(find /sys/devices -name press_to_select)
do
  dir=$(dirname $file)
  if [ -f $dir/speed -a -f $dir/sensitivity ]; then
    vTrackpointPath=$dir
    break
  fi
done

if ! [ $vTrackpointPath ]; then
  echo "Trackpoint not detected"
  exit 200
fi

## Initialize trackpoint varialbes with current values
vSensitivity=$(<$vTrackpointPath/sensitivity)
vSpeed=$(<$vTrackpointPath/speed)
vPress_to_Select=$(<$vTrackpointPath/press_to_select)
vScrolling="9"  # Initialize with 9 to allow test against numerical values

## Check if we need to set TrackPoint wheel
vTrackpointXconfig=$(xinput list-props "TPPS/2 IBM TrackPoint"  2> /dev/null | grep "Evdev Wheel Emulation (" | cut -d ' ' -f 1 | awk '{$1=$1};1')

## Determine and set correct path for X11 xorg.conf.d
if [ $vTrackpointXconfig ]; then
  echo "Searching for \"xorg.conf.d\". This may take a moment..."
  for root_dir in /usr /etc  # Note order precedence from least to greatest (here, the xorg.conf.d in "/etc" will be used
  do
    vXorgDir=$(find ${root_dir} -type d -name xorg.conf.d) &> /dev/null
  done
fi

## Update template with current values (to prevent settings being changed if user uses Optoin 3 to configure persistence)
sed -i "/vSensitivity=/c\vSensitivity=$vSensitivity" templates/trackpoint.sh
sed -i "/vSpeed=/c\vSpeed=$vSpeed" templates/trackpoint.sh
sed -i "/vPress_to_Select=/c\vPress_to_Select=$vPress_to_Select" templates/trackpoint.sh

while :
do
  ## Find and set xinput values for TrackPoint scrolling
  if [ $vTrackpointXconfig ]; then
    # Capture value for scrolling setting
    vTrackpointWheelEnabledValue=$(xinput list-props "TPPS/2 IBM TrackPoint" | grep "Evdev Wheel Emulation (" | cut -d ':' -f 2 | awk '{$1=$1};1')
    # Capture value for scrolling button
    vTrackpointWheelButtonValue=$(xinput list-props "TPPS/2 IBM TrackPoint" | grep "Evdev Wheel Emulation Button (" | cut -d ':' -f 2 | awk '{$1=$1};1')
    # Update template with current values to prevent them changing if persistence is enabled
    sed -i "s/\"EmulateWheel\".*/\"EmulateWheel\"   \"${vTrackpointWheelEnabledValue}\"/" templates/90-trackpoint.conf
    sed -i "s/\"EmulateWheelButton\".*/\"EmulateWheelButton\"   \"${vTrackpointWheelButtonValue}\"/" templates/90-trackpoint.conf
    # We don't worry about axis/horz. scrolling as genrally, if scrolling, then horz, else it's not applicable
  fi

  ## Check status of persistence
  systemctl --version &> /dev/null
  if [ $? != 0 ]; then
    vInitSystem=sysv
    if [ -f /etc/init.d/trackpoint -a -f /usr/bin/trackpoint.sh ]; then
      vInitStatus=Enabled
    elif [ ! -f /etc/init.d/trackpoint -a ! -f /usr/bin/trackpoint.sh ]; then
      vInitStatus=Disabled
    else
      vInitStatus='Broken. Use Option 3 or 5'
    fi
  else
    vInitSystem=systemd
    if [ -f /etc/systemd/system/trackpoint.service -a -f /etc/systemd/system/trackpoint.timer -a -f /usr/bin/trackpoint.sh ]; then
      vInitStatus=Enabled
    elif [ ! -f /etc/systemd/system/trackpoint.service -a ! -f /etc/systemd/system/trackpoint.timer -a ! -f /usr/bin/trackpoint.sh ]; then
      vInitStatus=Disabled
    else
      vInitStatus='Broken.  Use Option 3 or 5'
    fi
  fi

echo ""
echo "1: Check current Trackpoint settings"
echo "2: Set Trackpoint settings"
echo "3: Configure scrolling (only required for some GNU/Linux distributions)"
echo "4: Setup persistent Trackpoint settings using current settings"
echo "5: Make current Trackpoint settings persistent"
echo "6: Remove Trackpoint persistent settings (use OS defaults)"
echo "7: Set current Trackpoint settings back to OS defaults"
echo "0: Exit"
echo " "
read vMainMenu

case $vMainMenu in
1 )
  printf "Trackpoint Sensitivity: " && cat $vTrackpointPath/sensitivity
  printf "Trackpoint Speed: " && cat $vTrackpointPath/speed
  printf "Trackpoint Press_To_Select: " && cat $vTrackpointPath/press_to_select
  printf "Persistence: $vInitStatus ($vInitSystem)\n"
;;

2 )
  printf "Sensitivity (1-255): Currently " && cat $vTrackpointPath/sensitivity
  read vTempSensitivity
  if [ "$vTempSensitivity" -lt 1 -o "$vTempSensitivity" -gt 255 ]; then
    echo "Please enter a value between 1 and 255"
    continue
  fi
  printf "Speed (1-255): Currently " && cat $vTrackpointPath/speed
  read vTempSpeed
  if [ "$vTempSpeed" -lt 1 -o "$vTempSpeed" -gt 255 ]; then
    echo "Please enter a value between 1 and 255"
    continue
  fi
  printf "Press_to_Select (0-1): Currently " && cat $vTrackpointPath/press_to_select
  read vTempPress_to_Select
  if [[ "$vTempPress_to_Select" != [0-1] ]]; then
    echo "Please enter either 0 (disabled) or 1 (enabled)"
    continue
  fi
  ## Since all values are valid, assign to corrisponding variables
  vSensitivity=$vTempSensitivity
  vSpeed=$vTempSpeed
  vPress_to_Select=$vTempPress_to_Select
  echo $vSensitivity > $vTrackpointPath/sensitivity
  echo $vSpeed > $vTrackpointPath/speed
  echo $vPress_to_Select > $vTrackpointPath/press_to_select
  ## Update template with current values (to prevent settings being changed if user uses Optoin 3 to configure persistence)
  sed -i "/vSensitivity=/c\vSensitivity=$vSensitivity" templates/trackpoint.sh
  sed -i "/vSpeed=/c\vSpeed=$vSpeed" templates/trackpoint.sh
  sed -i "/vPress_to_Select=/c\vPress_to_Select=$vPress_to_Select" templates/trackpoint.sh
;;

3 )
  if [ $vTrackpointXconfig ]; then
    echo "Toggle scrolling (0 for \"No\", 1 for \"Yes\", [C]ancel): Currently ${vTrackpointWheelEnabledValue}"
    read vScrolling
    if [ $vScrolling = 0 ]; then
      xinput set-prop "TPPS/2 IBM TrackPoint" "Evdev Wheel Emulation" 0
      xinput set-prop "TPPS/2 IBM TrackPoint" "Evdev Wheel Emulation Button" 3
      xinput set-prop "TPPS/2 IBM TrackPoint" "Evdev Wheel Emulation Axes" 0 0 4 5
    elif [ $vScrolling = 1 ]; then
      xinput set-prop "TPPS/2 IBM TrackPoint" "Evdev Wheel Emulation" 1
      xinput set-prop "TPPS/2 IBM TrackPoint" "Evdev Wheel Emulation Button" 2
      xinput set-prop "TPPS/2 IBM TrackPoint" "Evdev Wheel Emulation Axes" 6 7 4 5  # Safe to assume horz. scrolling prefered
    else
      echo "Cancelled scrolling configuration"
    fi
  else
    echo "It doesn't appear wheel emulation is handled by evdev so skipping any configuration
    since it's assumed libinput genrally enables scrolling and this is the desired state"
  fi
;;

4 )
  if [ $vTrackpointXconfig ]; then
    echo "Would you like to include scrolling settings in persistence setup?
        (0 for \"No\", 1 for \"Yes\")"
    read vScrolling
    if [ $vScrolling = 1 ]; then
      echo "Copying config file to \"${vXorgDir}\""
      cp -r templates/90-trackpoint.conf $vXorgDir/90-trackpoint.conf
    fi
  fi

  if [ $vInitSystem = sysv ]; then
    cp -r templates/trackpoint /etc/init.d && chmod +x /etc/init.d/trackpoint
    cp -r templates/trackpoint.sh /usr/bin && chmod +x /usr/bin/trackpoint.sh
    update-rc.d trackpoint start 90 5 . stop 90 0 1 2 3 4 6 .
  else
    cp -r templates/trackpoint.service /etc/systemd/system
    cp -r templates/trackpoint.timer /etc/systemd/system
    cp -r templates/trackpoint.sh /usr/bin && chmod +x /usr/bin/trackpoint.sh
    systemctl daemon-reload
    systemctl start trackpoint
    systemctl enable trackpoint.timer
  fi
;;

5 )
if [ "$vInitStatus" = "Enabled" ]; then
  cp -r templates/trackpoint.sh /usr/bin && chmod +x /usr/bin/trackpoint.sh
  if [ $vScrolling = 1 ]; then
    cp -r templates/90-trackpoint.conf $vXorgDir/90-trackpoint.conf
  elif [ $vScrolling = 0 -a -f $vXorgDir/90-trackpoint.conf ]; then
    rm -f $vXorgDir/90-trackpoint.conf
  else
    echo "Skipping scrolling configuration"
  fi
else
  echo "Persistence not enabled.  Please run \"Option 4:\" first"
fi
;;

6 )
  if [ $vInitSystem = sysv -a $vInitStatus = Enabled ]; then
    update-rc.d -f trackpoint remove
    rm -f /etc/init.d/trackpoint
    rm -f /usr/bin/trackpoint.sh
    if [ -f $vXorgDir/90-trackpoint.conf ]; then
      rm -f $vXorgDir/90-trackpoint.conf
    fi
    ## Ensure current setting are not changed
    echo $vSensitivity > $vTrackpointPath/sensitivity
    echo $vSpeed > $vTrackpointPath/speed
    echo $vPress_to_Select > $vTrackpointPath/press_to_select
  elif [ $vInitSystem = systemd -a $vInitStatus = Enabled ]; then
    systemctl disable trackpoint.timer
    systemctl stop trackpoint &> /dev/null  # Output hidden since it warns about service being called by timer, but timer and service are removed below
    rm -f /etc/systemd/system/trackpoint.service
    rm -f /etc/systemd/system/trackpoint.timer
    rm -f /usr/bin/trackpoint.sh
    systemctl daemon-reload
    if [ -f $vXorgDir/90-trackpoint.conf ]; then
      rm -f $vXorgDir/90-trackpoint.conf
    fi
    ## Ensure current setting are not changed
    echo $vSensitivity > $vTrackpointPath/sensitivity
    echo $vSpeed > $vTrackpointPath/speed
    echo $vPress_to_Select > $vTrackpointPath/press_to_select
  else
    echo "Persistence not enabled.  Please run \"Option 4:\" first"
  fi
;;

7 )
echo 128 > $vTrackpointPath/sensitivity
echo 97 > $vTrackpointPath/speed
echo 0 > $vTrackpointPath/press_to_select
if [ $vScrolling < 9 ]; then
  echo "Reseting scrolling configuration not supported as it could potentially affect
  configurations not controlled by this script.  If you used this option, it's likely
  the default was disabled.  Run \"Optoin 3:\" and pass \"0\" to return to a disabled state"
fi
;;

0 | [eE]xit )
  exit
;;

* )
  echo "Pleaes enter a valid option"
esac
done
